lib/SimpleSAML/Utils/HTTP.php

```diff
@@ -85,6 +85,10 @@ private static function getServerHTTPS()
             return false;
         }
 
+        if ($_SERVER['HTTPS'] === null) {
```

Comment:
Is it possible for you to trace this somehow to see who and why is setting this variable to `null`? It doesn't make much sense to me...

In any case, this should be a separate pull request so that we can keep things clean, and the actual fix is to call `empty()` in the return statement, so that any value evaluated by PHP as empty makes the method return `false`, conforming to the PHP manual:

```php
return !empty($_SERVER['HTTPS']);
```